ac_new.rb

```diff
@@ -306,12 +306,48 @@ def ingest_to_kaltura(client, entry_name, meeting_id, sco_id, vid_file_path)
     results = client.upload_token_service.add(upload_token)
     upload_token_id = results.id
 
-    file_data = File.open(vid_file_path)
-    resume = false
-    final_chunk = true
-    resume_at = -1
+    if (File.size(vid_file_path) > 2*1024*1024*1024)
+    # chunked upload is required in this case.
+	dir=File.dirname(vid_file_path)
+	basename=File.basename(vid_file_path,'.mkv')
+
+	chunked_dir=File.join(dir, basename+'_chunked')
+
+	if File.exist?(chunked_dir)
+		system('rm -rf ' + chunked_dir)
+	end
+
+	Dir.mkdir(chunked_dir)
+	system('split -d -b 500m ' + vid_file_path + ' ' + File.join(chunked_dir, 'piece'))
```

Comment:
Just for FYI - you can split the file using Ruby native code, like so:
```ruby
file = File.open(vid_file_path)
i = 1
file.bytes.each_slice(50000000) do |slice|
  new_file = File.new("file_#{i}.chunk", "w+")
                      new_file.puts(slice.pack('C*'))
                      i = i + 1
end
```
However, considering the fact this code is not ever likely to run on non-UNIX OSes and that:
```
jess@jessex:/tmp$ time ruby /tmp/split.rb /path/to/file

real    2m2.694s
user    1m51.536s
sys     0m6.524s
jess@jessex:/tmp$ time split -d -b 50m /path/to/file /tmp/

real    0m2.711s
user    0m0.016s
sys     0m1.004s
```

We best stick to invoking the `split` util via `system()`